src/redisearch_rs/tracing_redismodule/src/lib.rs

```diff
@@ -0,0 +1,238 @@
+//!
+//! ## Example Syntax
+//!
+//! - `tokio::net=info` will enable all spans or events that:
+//!    - have the `tokio::net` target,
+//!    - at the level `info` or above.
+//! - `warn,tokio::net=info` will enable all spans and events that:
+//!    - are at the level `warn` or above, *or*
+//!    - have the `tokio::net` target at the level `info` or above.
+//! - `my_crate[span_a]=trace` will enable all spans and events that:
+//!    - are within the `span_a` span or named `span_a` _if_ `span_a` has the target `my_crate`,
+//!    - at the level `trace` or above.
+//! - `[span_b{name=\"bob\"}]` will enable all spans or event that:
+//!    - have _any_ target,
+//!    - are inside a span named `span_b`,
+//!    - which has a field named `name` with value `bob`,
+//!    - at _any_ level.
+//!
+//! ## Output Styling
+//!
+//! By default the subscriber will style the terminal output to help with legibility.
+//! To manually configure the styling of logging output you can set the `RUST_LOG_STYLE`
+//! environment variable. Supported values are:
+//!
+//! - `auto` (default) will attempt to print style characters, but don’t force the issue. If the console isn’t available on Windows, if TERM=dumb, or a CI environment is detected for example, then don’t print colors.
+//! - `always` will always print style characters even if they aren’t supported by the terminal. This includes emitting ANSI colors on Windows if the console API is unavailable.
+//! - `never` will never print style characters.
+//!
+//! [`level`]: tracing_core::Level
+//! [`Metadata`]: tracing_core::Metadata
+//! [`Targets`]: crate::filter::Targets
+//! [`env_logger`]: https://crates.io/crates/env_logger
+//! [`Filter`]: #impl-Filter<S>
+//! [global]: crate::layer#global-filtering
+//! [plf]: crate::layer#per-layer-filtering
+//! [filtering]: crate::layer#filtering-with-layers
+//! [re-syntax]: https://docs.rs/regex/1.11.1/regex/#syntax
+
+use std::cell::RefCell;
+use std::error::Error;
+use std::ffi::CStr;
+use std::ptr::NonNull;
+use std::{io, ptr};
+use tracing_core::LevelFilter;
+use tracing_subscriber::EnvFilter;
+use tracing_subscriber::fmt::MakeWriter;
+use tracing_subscriber::fmt::format::FmtSpan;
+
+/// Initializes a global subscriber that reports traces through `redismodule` logging.
+pub fn init(ctx: Option<NonNull<ffi::RedisModuleCtx>>) {
+    try_init(ctx).expect("Unable to install global tracing subscriber")
+}
+
+/// Initializes a global subscriber that reports traces through `redismodule`
+///  logging if one is not already set.
+///
+/// # Errors
+///
+/// Returns an Error if the initialization was unsuccessful, likely because
+/// a global subscriber was already installed by another call to `try_init`.
+pub fn try_init(
+    ctx: Option<NonNull<ffi::RedisModuleCtx>>,
+) -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
+    let env_filter = EnvFilter::builder()
+        .with_default_directive(LevelFilter::INFO.into())
```

Comment:
Will it be possible to get the `loglevel` Redis setting here instead for the default?

Reply:
Yes, you can set it in the config (I've been using it with search on disk). Although I can find any documentation on this. A quick search just gives this

https://linuxcommandlibrary.com/man/redis-server

Reply:
huh 🤔 good to know but yeah I'm really not sure this is exposed to the modules